node/test/timeouts.js

```diff
@@ -120,3 +120,219 @@ allocCluster.test('requests will timeout', {
+            (twoConn.options.timeoutFuzz / 2) + 1
+        ));
+
+        // timers module has weird semantics
+        // TODO: less magic
+        var newTime = timers.now() + adv;
+        timers.now = function now() {
+            return newTime;
+        };
+
+        timers.advance(adv);
+    }
+
+    function onTimeout(err) {
+        assert.equal(err && err.type, 'tchannel.request.timeout', 'expected timeout error');
+        assert.ok(err.logical, 'the timeout should be from logical request');
+        cluster.assertCleanState(assert, {
+            channels: [{
+                peers: [{
+                    connections: [
+                        {direction: 'in', inReqs: 0, outReqs: 0}
+                    ]
+                }]
+            }, {
+                peers: [{
+                    connections: [
+                        {direction: 'out', inReqs: 0, outReqs: 0}
+                    ]
+                }]
+            }]
+        });
+        assert.end();
+    }
+
+    function normalProxy(req, res, arg2, arg3) {
+        res.headers.as = 'raw';
+        res.sendOk(arg2, arg3);
+    }
+    function timeout(/* head, body, hostInfo, cb */) {
+        // do not call cb();
+    }
+});
+
+allocCluster.test('requests can succeed after timeout per attempt', {
+    numPeers: 3,
+    channelOptions: {
+        timers: timers
+    }
+}, function t(cluster, assert) {
+    var one = cluster.channels[0];
+    var two = cluster.channels[1];
+    var three = cluster.channels[2];
+    var sub = one.makeSubChannel({
+        serviceName: 'server'
+    });
+    var sub3 = three.makeSubChannel({
+        serviceName: 'server'
+    });
+
+    sub.register('/normal-proxy', normalProxy);
+    sub.register('/timeout', timeout);
+    sub3.register('/normal-proxy', normalProxy);
+    sub3.register('/timeout', timeout);
+
+    var twoSub = two.makeSubChannel({
+        serviceName: 'server',
+        peers: [
+            one.hostPort,
+            three.hostPort
+        ]
+    });
+
+    twoSub
+        .request({
+            serviceName: 'server',
+            hasNoParent: true,
+            headers: {
+                'as': 'raw',
+                cn: 'wat'
+            },
+            timeout: 1000
+        })
+        .send('/normal-proxy', 'h', 'b', onResp);
+
+    function onResp(err, res, arg2, arg3) {
+        assert.ifError(err);
+
+        assert.equal(String(arg2), 'h');
+        assert.equal(String(arg3), 'b');
+
+        twoSub
+            .request({
+                serviceName: 'server',
+                hasNoParent: true,
+                headers: {
+                    'as': 'raw',
+                    cn: 'wat'
+                },
+                timeout: 3000,
+                timeoutPerAttempt: 1000,
+                retryLimit: 2
+            })
+            .send('/timeout', 'h', 'b', done);
+    }
+
+    function done(err, res, arg2, arg3) {
+        assert.ifError(err);
+
```

Comment:
the logical `req` object contains an array outReqs

You should assert on `req.outReq`; namely that there are two out reqs and that the first failed with a timeout.